surveyor.py

```diff
@@ -5,477 +5,457 @@
+                        return raw_results
+                    
+                    elif self._edr in ['cortex', 'dfe', 's1']:
+                        self._results_collector = [] 
+                        for tag, results in product.get_results().items():
+                            self._results_collector.append(results)                    
+                            if len(self._results_collector ) > 0:
+                                log_echo(f"\033[92m-->query: {len(self._results_collector )} results \033[0m", self._log, use_tqdm=self._use_tqdm)
+                            else:
+                                log_echo(f"-->query: {len(self._results_collector )} results", self._log, use_tqdm=self._use_tqdm)
+                                
+                    if len(self._results_collector) > 1:
+                         return self._results_collector
+                    else:
+                        return "No results"
+                        
+                else:
+                    product.process_search(Tag('query'), base_query, query)
+                
+                for tag, results in product.get_results().items():
+                    self._write_results(results, query, "query", tag)
+            
+            # IOC
+            if ioc_list and (ioc_source and ioc_type):
+                # run search based on IOC file
 
+                    log_echo(f"Processing IOC list: {ioc_source}", self._log)
+                    ioc_list = [x.strip() for x in ioc_list]
+
+                    product.nested_process_search(Tag(f"IOC - {ioc_source}", data=ioc_source), {ioc_type: ioc_list}, base_query)
 
-table_template: Tuple[int, int, int, int, int, int] = (30, 30, 30, 30, 30, 30)
-table_template_str = f'{{:<{table_template[0]}}} ' \
-                     f'{{:<{table_template[1]}}} ' \
-                     f'{{:<{table_template[2]}}} ' \
-                     f'{{:<{table_template[3]}}}'
+                    for tag, results in product.get_results().items():
+                        self._write_results(results, ioc_source, 'ioc', tag)
+
+            # Sigma
+            sigma_rules = list()
+            if sigma_rule or sigma_rules or sigma_rules_str:
+                 
+                if sigma_rule:
+                    sigma_rules.append(sigma_rule)
+
+                #If list of one or more sigma rules is provided, rule(s) will be translated.
+                if sigma_rules:
+                    translated_rules = sigma_translation(product=self._edr, sigma_rules=sigma_rules, file=True)
+                    length = len(sigma_rules)
+
+            #If str of yaml containing one or more sigma rules is provided, rule(s) will be translated.
+                elif sigma_rules_str:
+                    translated_rules = sigma_translation(product=self._edr, sigma_rules=[sigma_rules_str], file=False)
+                    length = len(translated_rules['queries'])
```

Comment:
This is somewhat of a design question - do we want users to be able to specify all options: `sigma_rule`, `sigma_rules`, and `sigma_rules_str`? Or just one? For simplicity, I think the usage should be only one option is allowed.

How should we handle if more than one option is set? Which should take precedence? A small tweak on line 245 can set the precedence to be `sigma_rule` > `sigma_rules` > `sigma_rules_str`

```suggestion
                if sigma_rule:
                    sigma_rules.append(sigma_rule)

                #If list of one or more sigma rules is provided, rule(s) will be translated.
                elif sigma_rules:
                    translated_rules = sigma_translation(product=self._edr, sigma_rules=sigma_rules, file=True)
                    length = len(sigma_rules)

            #If str of yaml containing one or more sigma rules is provided, rule(s) will be translated.
                elif sigma_rules_str:
                    translated_rules = sigma_translation(product=self._edr, sigma_rules=[sigma_rules_str], file=False)
                    length = len(translated_rules['queries'])
```

Reply:
That makes sense, got a bit feature happy during the write. 

Tangent: Really wanted to break each mode out into their own functions. But to keep this update simple I opted against it. Definitely considering breaking those out, any reason we shouldn't in the future.

Reply:
All good! Agreed that this will work for the interim but there's room for improvement.

Another thought that we can save for a future PR - what if we simplified this even further so that `survey()` only accepts sigma_rules and definitions in dict form and `main()` handles all of the logic to determine if it's a folder of definitions vs. a single definition?

ex.
```
definition_files: dict[str, dict]
```